### PR TITLE
(Major) Implement doc & refactoring workflow demonstration

### PR DESCRIPTION
**Pull request by Demmonius**

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -11,6 +11,10 @@ export type VercelConfig = {
     public?: boolean
 }
 
+/**
+ * Official interface by Vercel can be find here: https://vercel.com/docs/api#endpoints/aliases/assign-an-alias-to-a-deployment
```
Comment (Vadorequest) [inline src/types.ts]: @see https://vercel.com/docs/api#endpoints/aliases/assign-an-alias-to-a-deployment

File: src/vercel.ts [hunk 2]
```diff
@@ -96,6 +98,10 @@ const create_aliases = async (deploymentUrl: string, customDeploymentFile: strin
 }
 
 const deploy = async (command: string, deployAlias: boolean, failIfAliasNotLinked: boolean): Promise<void> => {
+    /**
+     * Execute the command provided and store it into a variable
+     * exec_command also display the output
```
Comment (Vadorequest) [inline src/vercel.ts]: displays

File: .github/workflows/run-integration-test.yml [hunk 3]
```diff
@@ -20,14 +20,11 @@ jobs:
...
-        with:
-          command: ls
-      - uses: ./
-        with:
-          command: echo "test"
-        env:
-          VERCEL_TOKEN: "temporary nothing"
\ No newline at end of file
+          VERCEL_TOKEN: ${{ secrets.VERCEL_TOKEN }}
+          VERCEL_DEPLOYMENT_URL: "https://nrn-v2-mst-aptd-at-lcz-sty-c1.vercel.app"
```
Comment (Vadorequest) [inline .github/workflows/run-integration-test.yml]: Why providing `VERCEL_DEPLOYMENT_URL `?

File: .github/workflows/run-simple-integration-test.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# Summary:
+# Test the GitHub Action using an integration test.
+# Makes sure the GitHub Action works properly when running on a clean machine, without building anything (integration test).
+# This integration test will break if the Vercel project or deployment are deleted, or if the VERCEl_TOKEN doesn't have access to the deployment url.
```
Comment (Vadorequest) [inline .github/workflows/run-simple-integration-test.yml]: Remove useless line

File: .github/workflows/run-simple-integration-test.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      - 'releases/*'
+
+jobs:
+  run-integration-test:
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./
+        with:
+          command: "echo \"Deployed url: https://my-app.vercel.app\""
```
Comment (Vadorequest) [inline .github/workflows/run-simple-integration-test.yml]: This isn't really an integration test, shouldn't it rather really deploy something to vercel?

Even if it's a simple HTML page.

File: .github/workflows/run-integration-test.yml (hunk 3 above)
Comment (Vadorequest) [reply .github/workflows/run-integration-test.yml]: And why is that an url related to NRN? It shouldn't

**Review by Vadorequest — CHANGES_REQUESTED** [review]

File: .github/workflows/run-simple-integration-test.yml (hunk 5 above)
Comment (Demmonius) [reply .github/workflows/run-simple-integration-test.yml]: Okay it has been done

File: .github/workflows/run-integration-test.yml (hunk 3 above)
Comment (Demmonius) [reply .github/workflows/run-integration-test.yml]: This is how works the integration tests. You hardcode a vercel url you own, so you can apply aliases on it.

File: .github/workflows/run-example-deployment.yml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  run-example-deployment:
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          path: "github-action-deploy-on-vercel" # act doesn't work without a specific path. @see https://github.com/nektos/act/issues/228
+      - uses: ./
+        with:
+          command: "yarn deploy:example --token ${{ secrets.VERCEL_TOKEN }}"
+          apply_domain_aliases: true
```
Comment (Vadorequest) [inline .github/workflows/run-example-deployment.yml]: Why mix up both writing styles? 🤔 
Must be consistent.

**Review by Vadorequest — CHANGES_REQUESTED** [review]

File: .github/workflows/run-example-deployment.yml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    branches:
+      - '*'
+
+jobs:
+  run-example-deployment:
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+#        with:
+#          path: "github-action-deploy-on-vercel" # Act doesn't work if not specifying the path. See https://github.com/nektos/act/issues/228
```
Comment (Vadorequest) [inline .github/workflows/run-example-deployment.yml:14]: @Demmonius When specified, GHA fails. See https://github.com/UnlyEd/github-action-deploy-on-vercel/actions/runs/617433865

When commented, it passes.

What should we do about that? Is this supposed to be run through Act only? (locally)

Comment (Vadorequest) [reply .github/workflows/run-example-deployment.yml:14]: I'll keep it disabled for now until I get feedback from you

**Pull request merged**